src/core_modules/capture-core/rulesEngineActionsCreator/rulesEngineActionsCreator.js

```diff
@@ -30,18 +31,29 @@ export function getRulesActionsForEvent(
     foundation: ?RenderFoundation,
     formId: string,
     orgUnit: Object,
-    currentEventData: ?EventData  | {} = {},
+    currentEventData: ?EventData | {} = {},
     allEventsData: ?Array<EventData>,
 ) {
-    const rulesEffects = runRulesForSingleEvent(
-        rulesEngine,
-        program,
-        foundation,
-        orgUnit,
-        currentEventData,
-        allEventsData,
-    );
-    return getRulesActions(rulesEffects, foundation, formId);
+    const data = prepareEventData(program, foundation);
+    if (data) {
```

Comment:
This might look a bit strange and I agree it is. I will move this piece of code as it used to be hidden in a function in the following PR. It was mostly an experiment here :) Apologies